cromshell

```diff
@@ -402,12 +405,178 @@ function populateWorkflowIdAndServerUrl()
   fi
 }
 
+function _turtleSpinner()
+{
+  # Require a flag file to be given to this function.
+  # This keeps it safe.
+  if [[ $# -ne 1 ]] ; then
+    return 1
+  fi
+
+  local flagFile=$1
+
+  # Let's be fancy and make turtles that are walking while we wait:
+  local tmpTurtFile=$( makeTemp )
+  local turtFile=$( makeTemp )
+  local turtRFile=$( makeTemp )
+  turtle &> ${tmpTurtFile}
+  sed 's#^  ##g' ${tmpTurtFile} > ${turtFile}
+  rev ${turtFile} | sed -e 's$/$#$g' -e 's#\\#/#g' -e 's$#$\\$g' -e 's#(#)#g' -e 's#)#(#g' > ${turtRFile}
+
+  # Set up some loop variables:
+  local i=3
+  local incrementString="+1"
+  local curTurtFile=${turtFile}
+
+  # Let's loop while the ${flagFile} exists.
+  # Note that this will be spun off into its own thread:
+  while [ -f ${flagFile} ] ; do
```

Comment:
Why -f instead of -e?  Also, when do we need `[[`` vs `[`?

Reply:
`-f` tests that the input is a regular file and that it exists.  It's just a more specific test.

The single bracket syntax `[...]` is the traditional shell `test` syntax.  It can check a condition and set an error code.

The double bracket syntax `[[...]]` is similar, but is an "extended" test which allows for things like regex matching.  I've used them somewhat interchangeably in the code, but I don't have a good reason for it.  I tend to put more complex tests in double-braces and simple ones in single-braces.